src/trust.rs

```diff
@@ -146,6 +163,10 @@ pub fn create_device_certificate(
     Ok(())
 }
 
+pub fn verify_input_key(key_input: &str) -> Result<KeyPair> {
+    KeyPair::from_der(&read_from_der(key_input)).map_err(|e| anyhow!(e))
```

Comment:
read_from_der is calling `exit(1);` in case  of error, so why map the error ? You know it will exit anyway. Also, I think you can merge the two functions in one

Reply:
yes, actually this PR is more of rough work, because I wanted to ask about the `algo` thing. I would refractor it.